frontend/src/app/item/item.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, ViewChild } from '@angular/core';
+import { PageNotificationService } from '@nuvem/primeng-components';
+import { ItemService } from '../service/item.service';
+import { ItemCadastroComponent } from './item-cadastro/item-cadastro.component';
+import { ItemModel } from './model/item.model';
+
+
+@Component({
+  selector: 'app-item',
+  templateUrl: './item.component.html'
+})
+export class ItemComponent {
+  
+  itensCurr: ItemModel[];
+  @ViewChild(ItemCadastroComponent) itemCadastroComponent: ItemCadastroComponent;
+
+  constructor(
+    private itemService: ItemService, 
+    private notification: PageNotificationService
+    ) {}
+
+  buscarTodosHandler(){
+    this.buscarTodos();
+  }
+
+  buscarTodos(){
+    this.itensCurr = [];
+    this.itemService.listar().subscribe(
+      (itens: ItemModel[]) => {
+        for(let i = 0; i < itens.length; i++){
+          if(itens[i].idUsuario == (JSON.parse(localStorage.getItem("usuario")).id) as number){
```

Comment:
- JSON.parse(localStorage.getItem deveria estar em um service
- acesso a localStorage dentro de for
- `for` com contador é muito feio, substituir por `itens.forEach(item => {  ...  })`

Reply:
OK